src/object-to-ast.ts

```diff
@@ -0,0 +1,43 @@
+import type babelCore from '@babel/core'
+
+type ObjectToASTOptions = {
+  babel: typeof babelCore
+  fileOptions: babelCore.TransformOptions
+}
+
+export default function objectToAST(
+  object: unknown,
+  {babel, fileOptions}: ObjectToASTOptions,
+) {
+  const stringified = stringify(object)
+  const variableDeclarationNode = babel.template(`var x = ${stringified}`, {
+    preserveComments: true,
+    placeholderPattern: false,
+    ...fileOptions.parserOpts,
+    sourceType: 'module',
+  })() as babelCore.types.VariableDeclaration
+
+  return variableDeclarationNode.declarations[0].init
+}
+
+function stringify(object: unknown) {
+  // This check is necessary as `JSON.stringify` can return undefined,
+  // however the TypeScript definition for it does not include it.
+  // See https://github.com/microsoft/TypeScript/issues/18879 for details.
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  const str = JSON.stringify(object, stringifyReplacer) ?? 'undefined'
```

Comment:
this is changing runtime logic though, if object is `null`, the old version would throw an exception on `str.replace(...`, but now it will be `"undefined"` ... do we have permission to change this kind of stuff?

Reply:
wait,, this is stringify, not parse, nevermind

Reply:
fair point.  In practice though, if `object` is `null` then `JSON.stringify(object, stringifyReplacer)` will return `"null"` (as a string) as `null` is representable in `JSON`.  If you're very concerned, I'm happy to revert it.

Reply:
no no, it's fine as is - I didn't want to delete my comment outright because of email notification, so I just replied that I confused the problem with `JSON.parse` and marked conversation as resolved 😂

FTR: `JSON.parse(null)` would output a `null`, JSON.stringify(null) will output `"null"` so no problem here.